intelmq/bots/parsers/phishtank/parser.py

```diff
@@ -1,58 +1,63 @@
-# SPDX-FileCopyrightText: 2014 Tomás Lima
+# SPDX-FileCopyrightText: 2022 Filip Pokorný
 #
 # SPDX-License-Identifier: AGPL-3.0-or-later
 
-# -*- coding: utf-8 -*-
-import csv
-import io
 
-from intelmq.lib import utils
+import dateutil.parser
+
 from intelmq.lib.bot import ParserBot
 
 
 class PhishTankParserBot(ParserBot):
-    """Parse the PhishTank feed"""
-
-    def process(self):
-        report = self.receive_message()
-
-        columns = ["__IGNORE__",
-                   "source.url",
-                   "event_description.url",
-                   "time.source",
-                   "__IGNORE__",
-                   "__IGNORE__",
-                   "__IGNORE__",
-                   "event_description.target"
-                   ]
-
-        raw_report = utils.base64_decode(report.get("raw"))
-        raw_report = raw_report.translate({0: None})
-        for row in csv.reader(io.StringIO(raw_report)):
-
-            if not len(row):  # csv module can give empty lists
-                self.acknowledge_message()
-                return
-
-            # ignore headers
-            if "phish_id" in row:
-                continue
-
-            event = self.new_event(report)
-            event.change("feed.url", event["feed.url"][:event["feed.url"].find('data/')])
-
-            for key, value in zip(columns, row):
-
-                if key == "__IGNORE__":
-                    continue
-
-                event.add(key, value)
-
-            event.add('classification.type', 'phishing')
-            event.add("raw", ",".join(row))
-
-            self.send_message(event)
-        self.acknowledge_message()
+    """
+    Parse the PhishTank feed (json)
+    List of source fields:
+    [
+        'phish_id',
+        'url',
+        'phish_detail_url',
+        'submission_time',
+        'verified',
+        'verification_time',
+        'online',
+        'target',
+        'details'
+    ]
+    """
+
+    parse = ParserBot.parse_json
+    recover_line = ParserBot.recover_line_json
+
+    def parse_line(self, line, report):
+
+        event = self.new_event(report)
+
+        if line.get("submission_time"):
+            try:
+                event.add("time.source", str(dateutil.parser.parse(line.get("submission_time"))), raise_failure=False)
+            except dateutil.parser.ParserError:
+                self.logger.warning("Could not parse submission_time value '%s'", line.get("submission_time"))
+                pass
+
+        event.add("classification.type", "phishing")
+        event.add("extra.phishtank.phish_id", line.get("phish_id"), raise_failure=False)
+        event.add("source.url", line.get("url"))
+        event.add("event_description.url", line.get("phish_detail_url"), raise_failure=False)
+        event.add("event_description.target", line.get("target"), raise_failure=False)
+        event.add("status", "online", raise_failure=False)
```

Comment:
Before when I added it to `extra` field I didn't give it enough attention. But now I found in the docs this about the field:

_**online** Whether or not the phish is online and operational. In these data files, this will always be the string 'yes' since we only supply online phishes in these files._

Reply:
Fixed.